examples/arkode/CXX_manyvector/ark_sod_lsrk.hpp

```diff
@@ -0,0 +1,512 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Header file for ARKODE LSRKStep Sod shock tube example, see
+ * ark_sod_lsrk.cpp for more details.
+ * ---------------------------------------------------------------------------*/
+
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <sstream>
+#include <string>
+#include <vector>
+
+// Include desired integrators, vectors, linear solvers, and nonlinear solvers
+#include "arkode/arkode_erkstep.h"
+#include "arkode/arkode_lsrkstep.h"
+#include "nvector/nvector_manyvector.h"
+#include "nvector/nvector_serial.h"
+#include "sundials/sundials_core.hpp"
+
+// Macros for problem constants
+#define rhoL   SUN_RCONST(1.0)
+#define rhoR   SUN_RCONST(0.125)
+#define pL     SUN_RCONST(1.0)
+#define pR     SUN_RCONST(0.1)
+#define uL     SUN_RCONST(0.0)
+#define uR     SUN_RCONST(0.0)
+#define HALF   SUN_RCONST(0.5)
+#define ZERO   SUN_RCONST(0.0)
+#define ONE    SUN_RCONST(1.0)
+#define TWO    SUN_RCONST(2.0)
+#define FOURTH SUN_RCONST(0.25)
+
+#define NSPECIES 5
+
+#define WIDTH (10 + numeric_limits<sunrealtype>::digits10)
+
+using namespace std;
```

Comment:
This is considered a bad practice

Reply:
Perhaps, but it is used throughout many of our SUNDIALS examples.  Again, if we want to change that style, I recommend that it be deferred for a future PR, and that it be applied consistently throughout the library.

Reply:
I agree this should be moved to the `.cpp` file. It looks like a quick find an replace on `cout` is all that's needed here